src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2141,6 +2092,78 @@ RunDistributedExecution(DistributedExecution *execution)
 }
 
 
+/*
+ * UpdateWaitEventSet updates the waitEventSet for the distributed execution.
+ * This happens when the connection set for the distributed execution is changed,
+ * which means that we need to update which connections we wait on for events.
+ * It returns the new event set size.
+ */
+static int
+UpdateWaitEventSet(DistributedExecution *execution)
```

Comment:
Would be better to call it RebuildWaitEventSet, since this is more like a delete+insert than an update and there are cases in which we actuall update it.